extensions/amp-story/0.1/localization.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {closest} from '../../../src/dom';
+import {parseJson} from '../../../src/json';
+
+
+/**
+ * A unique identifier for each localized string.  Localized string IDs should:
+ *
+ *   - Maintain alphabetical order
+ *   - Be prefixed with the name of the extension that uses the string
+ *     (e.g. "AMP_STORY_"), or with "AMP_" if they are general
+ *   - NOT be reused; to deprecate an ID, comment it out and prefix its key with
+ *     the string "DEPRECATED_"
+ *
+ * Next ID: 22
+ *
+ * @const @enum {string}
```

Comment:
Just curious: any reason why we want strings instead of numbers?

Reply:
😦 

Yeah.  We directly use these as object keys in the `_locales/*.js` files, and object keys have to be strings.  If you let this be `@enum {number}` they get coerced to strings when used as keys, but then you get all kinds of nasty type errors when you try to use `Object.keys` because they're all strings, but anything that was accepting an ID is expecting a number.